charmhelpers/contrib/openstack/context.py

```diff
@@ -1032,6 +1033,13 @@ def configure_cert(self, cn=None):
         mkdir(path=ssl_dir)
         cert, key = get_cert(cn)
         if cert and key:
+            cert = b64decode(cert)
+            key = b64decode(key)
+            ca = config('ssl_ca')
+            ca = b64decode(ca) if ca else None
+            if not validate_cert(cert, key, ca):
+                return
```

Comment:
Returning here before configuring certs might be too drastic. There might be corner cases where the underlying library doesn't work as expected so we might be able to proceed by giving a warning for the time being.